src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -297,25 +298,57 @@ class ResolvePexConstraintsFile:
 class ResolvePexConfig:
     """Configuration from `[python]` that impacts how the resolve is created."""
 
+    indexes: tuple[str, ...]
+    find_links: tuple[str, ...]
+    manylinux: str | None
     constraints_file: ResolvePexConstraintsFile | None
     only_binary: tuple[str, ...]
     no_binary: tuple[str, ...]
 
+    def indexes_and_find_links_and_manylinux_pex_args(self) -> Iterator[str]:
+        # NB: In setting `--no-pypi`, we rely on the default value of `[python-repos].indexes`
+        # including PyPI, which will override `--no-pypi` and result in using PyPI in the default
+        # case. Why set `--no-pypi`, then? We need to do this so that
+        # `[python-repos].indexes = ['custom_url']` will only point to that index and not include
+        # PyPI.
+        yield "--no-pypi"
+        yield from (f"--index={index}" for index in self.indexes)
+        yield from (f"--find-links={repo}" for repo in self.find_links)
+
+        if self.manylinux:
+            yield "--manylinux"
+            yield self.manylinux
+        else:
+            yield "--no-manylinux"
+
 
 @dataclass(frozen=True)
 class ResolvePexConfigRequest(EngineAwareParameter):
     """Find all configuration from `[python]` that impacts how the resolve is created."""
 
-    resolve_name: str
+    resolve_name: str | None
```

Comment:
The behavior in this case isn't obvious (nor is the `debug_hint`). Would that compute a default value? Or would it only be valid if resolves were not configured?